pyvista/plotting/plotting.py

```diff
@@ -4720,13 +4724,12 @@ def _datasets(self):
 
     def __del__(self):
         """Delete the plotter."""
-        # We have to check here if it has the closed attribute as it
-        # may not exist should the plotter have failed to initialize.
-        if hasattr(self, '_closed'):
+        # We have to check here if the plotter was only partially initialized
+        if self._initialized:
             if not self._closed:
                 self.close()
         self.deep_clean()
-        if hasattr(self, 'renderers'):
+        if self._initialized:
             del self.renderers
```

Comment:
We were getting errors inside `self.close()` due to missing attributes, turned into warnings because they were raised in `__del__()` (I guess more plotters are dying now during the test suite).

Since errors can appear in many places during `__init__()`, plotters can have many levels of partial initialization. I've added an `_initialized` flag that is only `True` when `(Base)Plotter.__init__()` is done, and only do `self.deep_clean()` unless the plotter was fully initialized. This leaves some potential attributes that aren't cleaned (anything set before the error was raised), but this is probably fairly safe inside `__del__`: those resources should be released soon, hopefully.

Again I'm not 100% sure this is the right call, or the right implementation (e.g. set `self._initialized` to `True` at the end of `BasePlotter.__init__()`, but then to `False` again at the start of `Plotter.__init__()`). Test suite doesn't report VTK leaks at least, and the errors-turned-warnings-during-`__del__()` are now gone.